### ISSUE TITLE
Avoid puck blocking part of the popup

### ISSUE THREAD

**Issue by StarScape** [birchill/10ten-ja-reader#1833]
A negative downside of #1773 is that occasionally you run into a situation where you do a lookup, and if the word/kanji entry is big, the popup window will expand and the puck will be covering a portion of it. Like this:

![image](https://github.com/birchill/10ten-ja-reader/assets/4681820/c8b64b3f-f962-4d4e-b4a0-1a2bba16d269)

Before, you could just drag the puck out of the way. But now if you do that the event listener will detect that the puck is inside the 10ten popup and cause it to disappear.

My first thought was that the easy way to prevent this would be to just give the puck a lower `z-index` than the popup window. Thoughts on that?

**Comment by birtles** [comment]
> My first thought was that the easy way to prevent this would be to just give the puck a lower `z-index` than the popup window. Not sure if there is any conceivable downside to that. Thoughts?

Good question. It looks like we deliberately make the puck sit above the popup:

https://github.com/birchill/10ten-ja-reader/blob/f4d5d6c116fa459cdab9b0f1df136ffdfc063c5d/css/puck.css#L16-L21

If we made the popup cover the puck, would that be annoying? For example, you're scanning some text and it generates a massive popup which happens to cover the puck. Now you have to stop moving the puck, close the popup, and then start scanning again?

**Comment by StarScape** [comment]
Yeah, that’s a good point, I could see it interfering with scanning, which would be annoying. I think this might go back to the idea I discussed in #1767 of never letting the popup spawn too close to the puck in the first place.

For now, I think I’ll make a branch with the puck z-index dropped, run that locally for a few days, and see if it causes issues.

**Comment by birtles** [comment]
> For now, I think I’ll make a branch with the puck z-index dropped, run that locally for a few days, and see if it causes issues.

Sounds great.

I wonder if we will end up needing to do something complicated like dismissing the popup when the "moon" (smaller dot) passes over the popup and the "earth" (larger dot) is _not_ over the popup or something like that.

**Comment by StarScape** [comment]
I've been busy with interview-related stuff the past week so have been slow to follow up on this, but I got some time to play around with it this morning. Some notes:

> If we made the popup cover the puck, would that be annoying? For example, you're scanning some text and it generates a massive popup which happens to cover the puck. Now you have to stop moving the puck, close the popup, and then start scanning again?

This does happen from time to time with the z-index of the puck dropped. I could never get it to happen on initial lookup, i.e. in the Word tab, so thankfully it doesn't seem to interfere with scanning. But sometimes when I flip over to the Kanji or Names tabs the puck will be covered, and I'll have to close the popup and start over again.

I think this might still be better than the current behavior, where the _puck_ can cover part of the _popup_, with no way to fix it, since moving the puck causes the popup itself to disappear. However, I think we can still come up with a better solution.

> I wonder if we will end up needing to do something complicated like dismissing the popup when the "moon" (smaller dot) passes over the popup and the "earth" (larger dot) is not over the popup or something like that.

I'm intrigued by this idea. I might hack this together on my machine and give it a shot.

**Comment by birtles** [comment]
> I've been busy with interview-related stuff the past week so have been slow to follow up on this, but I got some time to play around with it this morning.

Great, thank you!

> > If we made the popup cover the puck, would that be annoying? For example, you're scanning some text and it generates a massive popup which happens to cover the puck. Now you have to stop moving the puck, close the popup, and then start scanning again?
> 
> This does happen from time to time with the z-index of the puck dropped. I could never get it to happen on initial lookup, i.e. in the Word tab, so thankfully it doesn't seem to interfere with scanning.

That's curious that it never happens on initial look up. I wonder why?

> > I wonder if we will end up needing to do something complicated like dismissing the popup when the "moon" (smaller dot) passes over the popup and the "earth" (larger dot) is not over the popup or something like that.
> 
> I'm intrigued by this idea. I might hack this together on my machine and give it a shot.

That would be great. Thank you!
